Shared/Supporting Files/Views/CategoryGridView.swift

```diff
@@ -0,0 +1,104 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+
+struct CategoryGridView: View {
+    /// All samples that will be shown in the categories.
+    private let samples: [Sample]
+    
+    /// The different sample categories generated from the samples list.
+    private let categories: [String]
+    
+    /// The columns for the grid.
+    private let columns = [
+        GridItem(.flexible()),
+        GridItem(.flexible())
+    ]
+    
+    init(samples: [Sample]) {
+        self.samples = samples
+        
+        var categoriesSet = Set<String>()
+        for sample in samples {
+            categoriesSet.insert(sample.category)
+        }
+        categories = Array(categoriesSet).sorted()
+    }
+    
+    var body: some View {
+        ScrollView {
+            LazyVGrid(columns: columns) {
+                ForEach(categories, id: \.self) { category in
+                    NavigationLink {
+                        SampleListView(samples: samples.filter { $0.category == category })
+                            .navigationTitle(category)
+                    } label: {
+                        CategoryTitleView(category: category)
+                    }
+                }
+            }
+            .padding(8)
+        }
+    }
+}
+
+private extension CategoryGridView {
+    struct CategoryTitleView: View {
+        /// The color scheme to detect dark mode.
+        @Environment(\.colorScheme) private var colorScheme
+        
+        /// The category name used to load the images from assets.
+        let category: String
+        
+        var body: some View {
+            Image("\(category.replacingOccurrences(of: " ", with: "-"))-bg")
+                .resizable()
+                .overlay {
+                    ZStack {
+                        Color.secondary
+                        Circle()
+                            .foregroundColor(.primary)
+                            .frame(width: 50, height: 50)
+                        Image("\(category.replacingOccurrences(of: " ", with: "-"))-icon")
+                            .colorInvert(colorScheme == .light)
+                        VStack {
+                            Spacer()
+                            Spacer()
+                            Spacer()
+                            Text(category)
+                                .foregroundColor(colorScheme == .dark ? .black : .white)
+                                .frame(maxWidth: .infinity)
+                            Spacer()
+                        }
```

Comment:
Instead of using multiple spacers, you can hardcode an y offset to the image.
```suggestion
                        Text(category)
                            .foregroundColor(colorScheme == .dark ? .black : .white)
                            .frame(maxWidth: .infinity)
                            .offset(y: 45)
```